src/components/__tests__/SearchBar.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { shallow } from "enzyme";
+import { createElement } from "react";
+import * as TestUtils from "react-dom/test-utils";
+
+import { SearchBar, SearchBarProps } from "../SearchBar";
+
+describe("SearchBar", () => {
+    const renderSearchBar = (props: SearchBarProps) => shallow(createElement(SearchBar, props));
+
+    it("renders structure correctly", () => {
+        const searchBarProps: SearchBarProps = {
+            defaultQuery: "",
+            onTextChangeAction:  jasmine.any(Function) as any,
+            placeHolder: "",
+            showSearchBar: true,
+            style: SearchBar.parseStyle("html{}")
+        };
+        const searchBar = renderSearchBar(searchBarProps);
+
+        expect(searchBar).toBeElement(
+            createElement("div",
+                {
+                    className: "widget-offline-search",
+                    style: SearchBar.parseStyle("{{...}}")
+                },
+                createElement("div", { className: "search-container" },
+                    createElement("span", { className: "glyphicon glyphicon-search" }),
+                    createElement("input", { className: "form-control", placeholder: "" }),
+                    createElement("button", { className: "btn-transparent" },
+                        createElement("span", { className: "glyphicon glyphicon-remove" })
+                    )
+                )
+            )
+        );
+    });
+
+    it("is not visible when show appearance is set to no", () => {
+        const searchBarProps: SearchBarProps = {
```

Comment:
It is possible for tests to run in parallel and in that case every data must be created by its own test to avoid issues